v8engine/src/main/java/cm/aptoide/pt/v8engine/social/presenter/TimelinePresenter.java

```diff
@@ -344,15 +356,58 @@ private void clickOnLikeSocialPost() {
             .filter(cardTouchEvent -> cardTouchEvent.getActionType()
                 .equals(CardTouchEvent.Type.LIKE) && cardTouchEvent.getCard()
                 .getType()
-                .isSocial())
-            .flatMapCompletable(cardTouchEvent -> timeline.like(cardTouchEvent.getCard(),
-                cardTouchEvent.getCard()
-                    .getCardId())))
+                .isSocial() || cardTouchEvent.getCard()
+                .getType()
+                .equals(CardType.MINIMAL_CARD))
+            .flatMapCompletable(cardTouchEvent -> accountManager.accountStatus()
+                .first()
+                .toSingle()
+                .flatMapCompletable(account -> {
+                  if (account.isLoggedIn()) {
+                    if (showCreateStore(account)) {
+                      return Completable.fromAction(
+                          () -> view.showCreateStoreMessage(SocialAction.LIKE));
+                    } else if (showSetUserOrStoreToPublic(account)) {
+                      return Completable.fromAction(() -> view.showSetUserOrStorePublicMessage());
+                    }
+
+                    final Post post = cardTouchEvent.getCard();
+                    return timeline.like(post, post.getCardId())
+                        .doOnCompleted(() -> view.updatePost(post,
+                            ((AppUpdateCardTouchEvent) cardTouchEvent).getCardPosition()));
```

Comment:
This throws a class cast exception because the cardTouchEvent might not be from an AppUpdate card. Reason why Likes in Social Posts don't work.